cpp/include/raft/core/nccl_clique.hpp

```diff
@@ -153,4 +145,4 @@ struct nccl_clique {
   std::vector<raft::device_resources> device_resources_;
```

Comment:
This is strange- device_resources has a nccl clique but a nccl clique has device resources? Please avoid the circular dependency here and just have the nccl_clique use the actua resources it needs (I assume that's just a stream?)